src/main/java/be/syntra/devshop/DevshopBack/services/utilities/CartMapper.java

```diff
@@ -1,45 +1,51 @@
 package be.syntra.devshop.DevshopBack.services.utilities;
 
-import be.syntra.devshop.DevshopBack.entities.Cart;
+import be.syntra.devshop.DevshopBack.entities.OrderContent;
+import be.syntra.devshop.DevshopBack.entities.ShopOrder;
 import be.syntra.devshop.DevshopBack.models.CartDto;
+import be.syntra.devshop.DevshopBack.models.CartProductDto;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
 import java.util.List;
+import java.util.stream.Collectors;
 
 import static java.util.stream.Collectors.toUnmodifiableList;
 
 @Component
 public class CartMapper {
+    private final ProductMapper productMapper;
 
+    @Autowired
+    public CartMapper(ProductMapper productMapper) {
+        this.productMapper = productMapper;
+    }
 
-    public CartDto convertToCartDto(Cart cart) {
-        return CartDto.builder()
-                .cartCreationDateTime(cart.getCartCreationDateTime())
-                .products(cart.getProducts())
-                .finalizedCart(cart.isFinalizedCart())
-                .paidCart(cart.isPaidCart())
+    public ShopOrder convertToCart(CartDto cartDto) {
+        return ShopOrder.builder()
+                .shopOrderCreationDateTime(cartDto.getCartCreationDateTime())
+                .finalizedShopOrder(cartDto.isFinalizedCart())
+                .paidShopOrder(cartDto.isPaidCart())
+                .orderContents(convertToCartContentList(cartDto.getCartProductDtoList()))
                 .build();
     }
 
-    public Cart convertToCart(CartDto cartDto) {
-        return Cart.builder()
-                .cartCreationDateTime(cartDto.getCartCreationDateTime())
-                .products(cartDto.getProducts())
-                .finalizedCart(cartDto.isFinalizedCart())
-                .paidCart(cartDto.isPaidCart())
-                .build();
+    private List<OrderContent> convertToCartContentList(List<CartProductDto> cartProductDtoList) {
+        return cartProductDtoList.stream()
+                .map(this::convertToCartContent)
+                .collect(Collectors.toList());
     }
 
-    public List<CartDto> convertToCartDtoList(List<Cart> carts) {
-        return carts.stream()
-                .map(this::convertToCartDto)
-                .collect(toUnmodifiableList());
+    private OrderContent convertToCartContent(CartProductDto cartProductDto) {
+        return OrderContent.builder()
+                .product(productMapper.convertToProduct(cartProductDto.getProductDto()))
+                .count(cartProductDto.getCount())
+                .build();
     }
 
-    public List<Cart> convertToCartList(List<CartDto> cartDtoList) {
+    List<ShopOrder> convertToCartList(List<CartDto> cartDtoList) {
```

Comment:
think about the encapsulation. Either we're consistent, and we make convertToCart package private. Or we convert this method to public. This class is an injectable component. So either are acceptable

Reply:
made it public to match other method, changed all method names to reference shopOrder in stead of cart

Reply:
renamed class to ShopOrderMapper